### DISCUSSION TITLE
new add-on groovyConsole (replace for wikdshell)

### DISCUSSION THREAD

**Discussion by EdoFro** [freeplane/freeplane#325] [Show and tell]
Hi all,
As mentioned in this [post](https://github.com/freeplane/freeplane/discussions/275#discussion-3964107), the **wikdshell** add-on  has some troubles running with Groovy 4.x.

## about wikdshell add-on
**wikdshell** is an add-on that opened a **Groovy Console** to write and use scripts with **Freeplane**.

**wikdshell** works with **Freeplane** versions up to **v1.9.15**.
But with the new versions starting as **v1.10.1** it doesn't work any longer.

## about the Groovy Console
The **Groovy Swing Console** allows a user to enter and run Groovy scripts. It is a standard component from the Groovy language.

It looks like this:

![Groovy Console](https://docs.groovy-lang.org/latest/html/documentation/assets/img/GroovyConsole.png)

more info about it can be found in the [Groovy Console official page](https://docs.groovy-lang.org/latest/html/documentation/groovy-console.html)

## about groovyConsole add-on
**groovyConsole** is a new add-on that opens a **Groovy Console** and works with **Groovy  4** and can be used with the **new versions** of **Freeplane**.

more info:
[Add-on repository in Github](https://github.com/EdoFro/Freeplane_groovyConsole)

Downloads:
[Release v0.0.1 (to download add-on file)](https://github.com/EdoFro/Freeplane_groovyConsole/releases/tag/v0.0.1)

Hope it helps,

edo

**Comment by EdoFro** [comment]
second version released:

* It has now access to the libraries
* Now it writes in the console or the log.0 file when using print() and println() 

can be downloaded from [here](https://github.com/EdoFro/Freeplane_groovyConsole/releases/tag/v0.0.2)

**Comment by EdoFro** [comment]
## v0.0.3
* bug corrections
* Added button 'starts System.out interceptor':
it restarts the systemOutInterceptor so that print() and println() get written in the log file or the console

can be downloaded from [here](https://github.com/EdoFro/Freeplane_groovyConsole/releases/tag/v0.0.3)

**Comment by EdoFro** [comment]
## v0.0.4
Added preference's parameters for Option Panel
 - show button's labels
 - default write to destination

Ok. I think this will be the last release for a long time. It works good enough for my needs.

It can be downloaded from [here ](https://github.com/EdoFro/Freeplane_groovyConsole/releases/tag/v0.0.4)

And I added more information about the add-on and their features in its repository:
https://github.com/EdoFro/Freeplane_groovyConsole

feedback is very much appreciated,
bye

edo

**Reply by sgjohnson1981** [reply]
@EdoFro left you an issue in your repo. Works well but hard to see when using a dark theme in FP.

**Reply by EdoFro** [reply]
Solved.

[version v0.0.8](https://github.com/EdoFro/Freeplane_groovyConsole/releases/tag/v0.0.8) released.

This is how it now looks when using a Dark Freeplane theme:

![groovyConsole dark themed](https://user-images.githubusercontent.com/35700575/258249694-37db6699-1339-4d6f-a7ae-dc673a6f719a.png)

**Reply by sgjohnson1981** [reply]
@EdoFro 
Was trying to play around with it myself. I noticed where the colors are hard-coded. Was going to see if I could at least just hard-code what I wanted them set as, but I noticed in groovyConsole.mm there's a couple nodes under zips (doc and templates), that aren't in the repo. Can I just delete those nodes, or are they important?

**Reply by sgjohnson1981** [reply]
Also noticed the updateUrl attribute is different in groovyConsole.mm in the repo versus the finished add-on in releases. The former uses {homepage} but the latter uses the entire URL. When I went to package mine after modifying and building I got a malformedURLException when I tried to install it. Did I build mine wrong somehow?

**Reply by EdoFro** [reply]
I use this version of the [devtoolsAddon](https://github.com/EdoFro/freeplane-devtools)

This version translates some of the variables when packaging that are not translated by the addon installer.

**Reply by EdoFro** [reply]
latest release:
https://github.com/EdoFro/freeplane-devtools/releases/tag/v0.10.1

**Reply by nnako** [reply]
hi @EdoFro , will you be considered the "new" maintainer for the `freeplane-devtools` addon? `gpapp`'s last commit happened about 14 months ago. Or will your changes be merged to his repo, eventually?

to summarize all the locations I found:

- [in the Addons folder within Freeplane's project on GitHub](https://github.com/freeplane/addons/tree/master/devtools)
-> contains the installable addon
-> Addon v0.9.27 for FP v1.7.4+

- [in SourceForge's download section](https://sourceforge.net/projects/freeplane/files/addons/devtools/)
-> contains the installable addon
-> Addon v0.9.27 for FP v1.7.4+

- [in `gpapp`'s GitHub repo for this addon](https://github.com/gpapp/freeplane-devtools) 
-> installable Addon v0.9.31
-> development version of Addon v0.9.31

- [in `EdoFro`'s GitHub repo for this addon](https://github.com/EdoFro/freeplane-devtools)
-> installable Addon v0.10.1
-> latest development version of Addon

**Reply by EdoFro** [reply]
hi @nnako,

> will you be considered the "new" maintainer for the freeplane-devtools addon?

ok. I use it and maintain it for myself, so I can solve bugs and answer questions for other users if needed.

I prefer to use my repo, to use its issues and discussion panels and have direct control over them.

I added more features that I wanted to simplify the process of creating addons and I think this features might be helpful to others too. 

BR

edo

**Comment by EdoFro** [comment]
> Can I just delete those nodes, or are they important?


 You can. They are not important. I put them just in case. I have no files in those folders in my repo neither
